src/Products/Annotation/Controllers/AnnotationApiController.cs

```diff
@@ -73,357 +55,214 @@ public AnnotationConfiguration LoadConfig()
             }
+
+            description.guid = documentGuid;
+            // return document description
+            return description;
         }
 
         /// <summary>
-        /// Download document
+        /// Get all annotations from the document
         /// </summary>
-        /// <param name="path">string</param>
-        /// <param name="annotated">bool</param>
-        /// <returns></returns>
-        [HttpGet]
-        [Route("downloadDocument")]
-        public HttpResponseMessage DownloadDocument(string path)
+        /// <param name="documentGuid">string</param>
+        /// <param name="documentType">string</param>
+        /// <returns>AnnotationInfo[]</returns>
+        private AnnotationBase[] GetAnnotations(string documentGuid, string documentType, string password)
         {
-            // prepare response message
-            HttpResponseMessage response = new HttpResponseMessage(HttpStatusCode.OK);          
-           
-            // add file into the response
-            if (File.Exists(path))
+            try
             {
-                
-                var fileStream = GetCleanDocumentStream(path);
-                response.Content = new StreamContent(fileStream);
-                response.Content.Headers.ContentType = new MediaTypeHeaderValue("application/octet-stream");
-                response.Content.Headers.ContentDisposition = new ContentDispositionHeaderValue("attachment")
+                FileStream documentStream = new FileStream(documentGuid, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
+                DocumentType docType = DocumentTypesConverter.GetDocumentType(documentType);
+                AnnotationBase[] annotations;
+
+                using (GroupDocs.Annotation.Annotator annotator = new GroupDocs.Annotation.Annotator(documentStream, GetLoadOptions(password, true)))
                 {
-                    FileName = Path.GetFileName(path)
-                };
-                return response;
+                    annotations = annotator.Get().ToArray();
+                }
+
+                documentStream.Dispose();
+                documentStream.Close();
+                return annotations;
             }
-            else
+            catch (System.Exception ex)
             {
-                return new HttpResponseMessage(HttpStatusCode.NotFound);
+                throw ex;
             }
         }
 
-        /// <summary>
-        /// Download document
-        /// </summary>
-        /// <param name="path">string</param>
-        /// <param name="annotated">bool</param>
-        /// <returns></returns>
-        [HttpPost]
-        [Route("downloadAnnotated")]
-        public HttpResponseMessage DownloadAnnotated(AnnotationPostedDataEntity annotateDocumentRequest)
+        private List<string> GetAllPagesContent(string password, string documentGuid, IDocumentInfo pages)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make 'GetAllPagesContent' a static method.](https://app.codacy.com/gh/groupdocs-annotation/GroupDocs.Annotation-for-.NET-MVC/pullRequest?prid=5372471)